packages/entities/entities-plugins/src/components/free-form/shared/RedisSelector.vue

```diff
@@ -0,0 +1,407 @@
+            </div>
+          </template>
+        </KSelect>
+      </div>
+      <RedisConfigCard
+        v-if="selectedRedisConfig"
+        :config-fields="selectedRedisConfig"
+      />
+      <p
+        v-if="sharedRedisConfigFetchError"
+        class="redis-shared-config-error-message"
+        data-testid="redis-config-fetch-error"
+      >
+        {{ sharedRedisConfigFetchError || t('redis.shared_configuration.error') }}
+      </p>
+    </div>
+    <ObjectField
+      v-else
+      v-bind="props"
+      as-child
+      :fields-order="fieldsOrder"
+      :name="formRedisPath"
+      reset-label-path="reset"
+    />
+  </KCard>
+  <ObjectField
+    v-else
+    v-bind="props"
+    :fields-order="fieldsOrder"
+    hide-required-asterisk
+    :name="formRedisPath"
+  />
+</template>
+
+<script setup lang="ts">
+import ObjectField from '../shared/ObjectField.vue'
+import RedisConfigCard from './RedisConfigCard.vue'
+import { onBeforeMount, inject, computed, ref, watch, type Ref } from 'vue'
+import english from '../../../locales/en.json'
+import { createI18n } from '@kong-ui-public/i18n'
+import { KUI_ICON_SIZE_20 } from '@kong/design-tokens'
+import { FORMS_CONFIG, REDIS_PARTIAL_FETCHER_KEY, REDIS_PARTIAL_INFO } from '@kong-ui-public/forms'
+import { AddIcon } from '@kong/icons'
+import type { SelectItem } from '@kong/kongponents/dist/types'
+import { useAxios, useDebouncedFilter, useErrors, type KongManagerBaseFormConfig, type KonnectBaseFormConfig } from '@kong-ui-public/entities-shared'
+import type { RedisConfig, RedisPartialType, Redis } from './types'
+import { getRedisType, getPartialTypeDisplay } from './utils'
+import { useField, useFormData } from './composables'
+defineEmits<{
+  (e: 'showNewPartialModal'): void
+}>()
+
+const { t } = createI18n<typeof english>('en-us', english)
+
+const fieldsOrder = [
+  'host', 'port', 'connection_is_proxied', 'database', 'username', 'password',
+  'sentinel_master', 'sentinel_role', 'sentinel_nodes', 'sentinel_username', 'sentinel_password',
+  'cluster_nodes', 'cluster_max_redirections', 'ssl', 'ssl_verify', 'server_name',
+  'keepalive_backlog', 'keepalive_pool_size', 'read_timeout', 'send_timeout', 'connect_timeout',
+]
+const endpoints = {
+  konnect: {
+    getOne: '/v2/control-planes/{controlPlaneId}/core-entities/partials/{id}',
+    getAll: '/v2/control-planes/{controlPlaneId}/core-entities/partials',
+  },
+  kongManager: {
+    getOne: '/{workspace}/partials/{id}',
+    getAll: '/{workspace}/partials',
+  },
+}
+interface RedisSelectorProps {
+  isEditing?: boolean
+  defaultRedisConfigItem?: string
+  redisType?: RedisPartialType
+  redisPath?: string
+}
+
+type PartialArray = Array<{ id: string | number, path?: string | undefined }>
+
+type PartialInfo = {
+  redisType: Ref<RedisPartialType>
+  redisPath: Ref<string>
+}
+
+const props = defineProps<RedisSelectorProps>()
+
+const redisPartialFetcherKey: Ref<number, number> | undefined = inject(REDIS_PARTIAL_FETCHER_KEY)
```

Comment:
You can set a default value to inject to avoid the warning when the value is not provided.